File: 04.Simulation模拟/leetcode_3612_processStr.py
class Solution:
    def processStr(self, s: str) -> str:
        res = ""
        for i in s:
            if i == "*":
                res = res[:-1]
            elif i == "#":
                res += res
            elif i == "%":
                res = res[::-1]
            else:
                res += i
        return res

if __name__ == "__main__":
    sl = Solution()
    s = "a#b%*"
    print(sl.processStr(s))

File: 04.Simulation模拟/leetcode_3612_processStr.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    string processStr(string s) {
        string res = "";
        for (auto& c : s){
            if (c == '*'){
                res = res.substr(0, res.size()-1);
            }
            else if (c == '#'){
                res += res;
            }
            else if (c == '%'){
                string cur = "";
                for (auto& i : res){
                    cur = i+cur;
                }
                res = cur;
            }
            else{
                res += c;
            }
        }
        return res;
    }
};

int main(){
    Solution sl;
    string s = "a#b%*";
    cout << sl.processStr(s) << endl;
}
